services/scraper/scraper.go

```diff
@@ -0,0 +1,140 @@
+package scraper
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+
+	"github.com/gocolly/colly"
+	"github.com/gocolly/colly/queue"
+)
+
+var selectors = map[string]string {
+	"wholePage": "html",
+	"topImageAds": "#tvcap .plantl a.pla-unit-title-link",
+	"topLinkAds": "#tvcap .d5oMvf > a",
+	"sideImageAds": ".cu-container a.plantl.clickable-card",
+	"nonAds": "#search .yuRUbf > a",
+	"bottomAds": "#bottomads .uEierd > a",
+}
+
+// TODO: Replace this struct with model
+type Result struct {
+	Keyword string
+	TopAdLinks []string
+	OtherAdLinks []string
+	NonAdLinks []string
+	PageCache string
+}
+var results = map[string]*Result{}
+
+const GOOGLE_SEARCH_URL = "http://www.google.com/search?q=%s"
+
+func Search(keywords []string) {
+	collector := colly.NewCollector(colly.Async(true))
+
+	q, err := queue.New(2, &queue.InMemoryQueueStorage{MaxSize: 10000})
+	if err != nil {
+		log.Fatal("Failed to create a queue", err.Error())
+	}
+
+	for _, keyword := range keywords {
```

Comment:
Based on our new convention, loops variable should be single line letter
```suggestion
	for _, k := range keywords {
```